src/main/java/org/pawaneuler/Generator/AssociationRuleGenerator/AssociationRuleGenerator.java

```diff
@@ -0,0 +1,232 @@
+package org.pawaneuler.Generator.AssociationRuleGenerator;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import org.pawaneuler.DataTypes.Trie.*;
+import org.pawaneuler.DataTypes.Rule.*;
+
+/**
+ * @author marissanura
+ */
+public class AssociationRuleGenerator {
+    private AssociationRule AR;
+    private Trie T;
+    private Frequency frequency;
+
+    /**
+     * 
+     * @param T trie that want to generate to Association Rule
+     */
+    public AssociationRuleGenerator(Trie T) {
+        this.AR = new AssociationRule();
+        this.T = T;
+        this.frequency = new Frequency(T);
+    }
+
+    /**
+     * 
+     * @param T trie that want to generate to Association Rule
+     * @param minSup minimum support for generating Association Rule
+     */
+    public AssociationRuleGenerator(Trie T, int minSup) {
+        this.AR = new AssociationRule(minSup);
+        this.T = T;
+        this.frequency = new Frequency(T);
+    }
+
+    /**
+     * to execute the Association Rule Generation from a Trie
+     */
+    public void execute() {
+        frequency.generateAllFreq();
+        ArrayList<ArrayList<String>> allFreqSet = generateFrequentItemset();
+
+        for (ArrayList<String> subset : allFreqSet) {
+            AR.getAssociationRules().addAll(generatRuleFromSubset(subset));
+        }
+
+        AR.sort();
+    }
+
+    /**
+     * to print all association rule
+     */
+    public void printAllAssociationRules() {
+        this.AR.printAll();
+    }
+
+    /**
+     * 
+     * @return 2-dimentional ArrayList from frequent itemset
+     */
+    private ArrayList<ArrayList<String>> generateFrequentItemset() {
+        int kItemset = 1; // Combinations start with 1-itemset
+        ArrayList<ArrayList<String>> frequentItemset = new ArrayList<ArrayList<String>>();
+        ArrayList<String> candidateSingleset = generateProducts(); //Unique singleset for frequent itemset
+        ArrayList<ArrayList<String>> candidate = combination(candidateSingleset,kItemset); //Frequent Itemset candidate
+
+        do {
+            kItemset++;
+            pruning(candidate);
+            candidateSingleset = getUniqueProduct(candidate);
+            candidate = combination(candidateSingleset,kItemset);
+            frequentItemset.addAll(candidate);
+        } while (candidate.size() > kItemset);
+
+        return frequentItemset;
+    }
+
+    /**
+     * 
+     * @return ArrayList of trie products
+     */
+    private ArrayList<String> generateProducts() {
+        ArrayList<String> products = new ArrayList<String>();
+        ArrayList<Node> nodes = T.getNodes();
+        for (Node node : nodes) {
+            if (node.isRoot()) {
+                continue;
+            }
+            else {
+            //Make sure there's no duplicate
+            products.remove(node.getProduct());
+            products.add(node.getProduct());
```

Comment:
Make all code with explanation `//Make sure there's no duplicate` a function, something in the name of `insertUniqueItemIntoArrayList`.